test/libsolidity/semanticTests/operators/compound_assign_transient_storage.sol

```diff
@@ -0,0 +1,23 @@
+contract test {
+	uint transient value1;
+	uint transient value2;
+	function f(uint x, uint y) public returns (uint w) {
+		uint value3 = y;
+		value1 += x;
+		value3 *= x;
+		value2 *= value3 + value1;
+		return value1 += 7;
+	}
+}
```

Comment:
Wasn't this supposed to return `value2`? The way it is now, most of these calculations have no effect so even if they're wrong we don't see that.

Also, I'd actually include both storage and transient storage variables here. Generally, one kind of bug I'd be worried about would be a helper for both getting the same name and only one version being included (this is more likely to happen in IR codegen but these tests will cover IR too). Or wrong specialization being used. That's the reason I suggested mixing them in a single test in some of those cases.

Reply:
Yeah, I just copied this test which used storage variables and forget to mix and adjust everything. I will fix that shortly.

Reply:
I made the changes requested and checked that everything is being computed as expected. `value1` (now a storage variable) increases as `f` is called and `value2` is always reset.